presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -716,6 +756,66 @@ public void testSchemaAuthorization()
         assertUpdate(admin, "DROP ROLE admin");
     }
 
+    @Test
+    public void testShowCreateSchema()
+    {
+        Session admin = Session.builder(getQueryRunner().getDefaultSession())
+                .setIdentity(Identity.forUser("hive")
+                        .withRole("hive", new SelectedRole(ROLE, Optional.of("admin")))
+                        .build())
+                .build();
+
+        Session user = testSessionBuilder()
+                .setCatalog(getSession().getCatalog().get())
+                .setSchema("test_show_create_schema")
+                .setIdentity(Identity.forUser("user").withPrincipal(getSession().getIdentity().getPrincipal()).build())
+                .build();
+
+        assertUpdate(admin, "CREATE ROLE test_show_create_schema_role");
+        assertUpdate(admin, "GRANT test_show_create_schema_role TO user");
+
+        assertUpdate(admin, "CREATE SCHEMA test_show_create_schema");
+
+        String createSchemaSql = format("" +
+                "CREATE SCHEMA %s.test_show_create_schema\n" +
+                "AUTHORIZATION USER hive",
+                getSession().getCatalog().get());
+
+        String actualResult = getOnlyElement(computeActual(admin, "SHOW CREATE SCHEMA test_show_create_schema").getOnlyColumnAsSet()).toString();
+        assertEquals(actualResult, createSchemaSql);
+
+        assertQueryFails(user, "SHOW CREATE SCHEMA test_show_create_schema", "Access Denied: Cannot show create schema for test_show_create_schema");
+
+        assertUpdate(admin, "ALTER SCHEMA test_show_create_schema SET AUTHORIZATION ROLE test_show_create_schema_role");
+
+        createSchemaSql = format("" +
+                "CREATE SCHEMA %s.test_show_create_schema\n" +
+                "AUTHORIZATION ROLE test_show_create_schema_role",
+                getSession().getCatalog().get());
+
+        actualResult = getOnlyElement(computeActual(admin, "SHOW CREATE SCHEMA test_show_create_schema").getOnlyColumnAsSet()).toString();
+        assertEquals(actualResult, createSchemaSql);
+
+        // test that LOCATION is shown when it does not match the metastore's default
+        testingMetastore.overrideConfig(WAREHOUSE.getVarname(), "x");
```

Comment:
You should use io.prestosql.testing.AbstractTestQueryFramework#executeExclusively